sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/policy/RetryPolicy.java

```diff
@@ -122,14 +152,13 @@ private HttpResponse attempt(final HttpRequest httpRequest, final HttpPipelineNe
         try {
             httpResponse = next.clone().process();
         } catch (RuntimeException err) {
-            if (shouldRetryException(retryStrategy, err, tryCount)) {
+            if (shouldRetryException(retryStrategy, err, tryCount, suppressed)) {
                 logRetryWithError(LOGGER.atVerbose(), tryCount, "Error resume.", err);
 
                 try {
                     Thread.sleep(retryStrategy.calculateRetryDelay(tryCount).toMillis());
                 } catch (InterruptedException ie) {
-                    err.addSuppressed(ie);
-                    throw LOGGER.logThrowableAsError(err);
+                    throw LOGGER.logThrowableAsError(new RuntimeException(ie));
```

Comment:
I think this should be reverted. This was changed recently because if we did have an interrupt we'd drop the initial exception that caused the retry to trigger.